### PR TITLE
Fixes protected roles getting antagonist

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Because we can't be trusted to keep dynamic.json up to date with roles, we're just going to do it the easy way.

Jobs now have an antagonist_restricted flag which simply prevents the role from rolling a set list of antagonists. If no list is set, then it will assume all antags.

closes https://github.com/Skyrat-SS13/Skyrat-tg/pull/11000

## Changelog

Not player facing.

### CODE REVIEW

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 1]
```diff
@@ -137,30 +118,10 @@
...
-		JOB_SCIENCE_GUARD,
-		JOB_VANGUARD_OPERATIVE,
-		JOB_PRISONER,
-		// SKYRAT EDIT END
 	)
 	required_candidates = 2
-	weight = 4
-	cost = 15
+	weight = 2
+	cost = 12
```
Comment (GoldenAlpharex) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:124]: Why's the weight and cost being changed here?

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Also, if antag_flag is actually a bitflag, you should use that to make a `restricted_antag_flags` var instead of using a list, as that would be both faster, cleaner and easier to read.

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm (hunk 1 above)
Comment (Gandalf2k15) [reply code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:124]: That file was reset to TG.

**Review by GoldenAlpharex — APPROVED** [review]
Looks good, I'll test-merge it first to be sure we don't end up with everyone unable to be antag.

File: code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm [hunk 2]
```diff
@@ -16,6 +16,22 @@
...
+		// SKYRAT EDIT ADDITION - PROTECTED JOBS
+		else if(P.mind.assigned_role.antagonist_restricted)
+			if(P.mind.assigned_role.restricted_antagonists)
+				if(antag_flag in P.mind.assigned_role.restricted_antagonists)
+					candidates.Remove(P)
+					continue
+			else //Assume it's all antagonist roles.
+				candidates.Remove(P)
+				continue
+		else if(!P.client.prefs?.read_preference(/datum/preference/toggle/be_antag))
```
Comment (tf-4) [inline code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm]:
```suggestion
		else if(P.client?.prefs && !P.client.prefs.read_preference(/datum/preference/toggle/be_antag))
```

File: code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm [hunk 3]
```diff
@@ -16,6 +16,22 @@
...
+			candidates.Remove(P)
+			continue
+		else if(is_banned_from(P.client.ckey, BAN_ANTAGONIST))
```
Comment (tf-4) [inline code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm]:
```suggestion
		else if(is_banned_from(P.client?.ckey, BAN_ANTAGONIST))
```

**Comment by Gandalf2k15** [comment]
tests show this works well

**Pull request merged**